dotnet/src/SemanticKernel.Abstractions/Search/TextSearchResult.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Search;
+
+/// <summary>
+/// Represents a search result retrieved from a <see cref="ITextSearch" /> instance.
+/// </summary>
+/// <remarks>
+/// An instance of <see cref="TextSearchResult"/> is a normalised search result which provides access to:
+/// - Name associated with the search result
+/// - Value associated with the search result
+/// - Link reference associated with the search result
+/// </remarks>
+[Experimental("SKEXP0001")]
+public class TextSearchResult(string? name = null, string? value = null, string? link = null, object? innerResult = null)
```

Comment:
I wonder if it would make sense to name this class more for what it contains rather than what produces it.  E.g. something like LinkedTextContent.  Since text search can produce many different types of result already, calling this one TextSearchResult seems overly broad.

Reply:
@westey-m I have changed this to `TextSearchContent`

Reply:
Follow up task: https://github.com/microsoft/semantic-kernel/issues/8338